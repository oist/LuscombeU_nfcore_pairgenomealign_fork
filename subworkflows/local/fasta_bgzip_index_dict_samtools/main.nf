include { SAMTOOLS_BGZIP     } from '../../../modules/nf-core/samtools/bgzip/main'
include { SAMTOOLS_DICT      } from '../../../modules/nf-core/samtools/dict/main'
include { SAMTOOLS_FAIDX     } from '../../../modules/nf-core/samtools/faidx/main'

workflow FASTA_BGZIP_INDEX_DICT_SAMTOOLS {

    take:
    ch_fasta // channel: [ val(meta), fasta ]

    main:

    ch_versions = Channel.empty()

    // Guarantee BGZIP compression
    SAMTOOLS_BGZIP ( ch_fasta )
    ch_versions = ch_versions.mix(SAMTOOLS_BGZIP.out.versions)

    SAMTOOLS_FAIDX ( SAMTOOLS_BGZIP.out.fasta, [[],[]], [[],[]] )
    ch_versions = ch_versions.mix(SAMTOOLS_FAIDX.out.versions)

    SAMTOOLS_DICT  ( SAMTOOLS_BGZIP.out.fasta )
    ch_versions = ch_versions.mix(SAMTOOLS_DICT .out.versions)

    emit:
    fasta_gz = SAMTOOLS_BGZIP.out.fasta        // channel: [ val(meta),  fasta.gz ]
    fai      = SAMTOOLS_FAIDX.out.fai          // channel: [ val(meta),  fai ]
    gzi      = SAMTOOLS_FAIDX.out.gzi          // channel: [ val(meta),  gzi ]
    dict     = SAMTOOLS_DICT .out.dict         // channel: [ val(meta),  dict ]

    versions = ch_versions                     // channel: [ versions.yml ]
}
