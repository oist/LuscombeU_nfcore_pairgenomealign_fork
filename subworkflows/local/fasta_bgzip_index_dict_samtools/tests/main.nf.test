nextflow_workflow {

    name "Test Subworkflow FASTA_BGZIP_INDEX_DICT_SAMTOOLS"
    script "../main.nf"
    workflow "FASTA_BGZIP_INDEX_DICT_SAMTOOLS"

    tag "subworkflows"
    tag "subworkflows_"
    tag "subworkflows/fasta_bgzip_index_dict_samtools"
    tag "samtools"
    tag "samtools/bgzip"
    tag "samtools/dict"
    tag "samtools/faidx"

    test("sarscov2 - fasta") {

        when {
            workflow {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }
}
