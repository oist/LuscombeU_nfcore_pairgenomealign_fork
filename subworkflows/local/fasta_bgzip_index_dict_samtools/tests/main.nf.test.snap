{
    "sarscov2 - fasta": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fasta.gz:md5,6e9fe4042a72f2345f644f239272b7e6"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fasta.gz.fai:md5,9da2a56e2853dc8c0b86a9e7229c9fe5"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fasta.gz.gzi:md5,7dea362b3fac8e00956a4952a3d4f474"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fasta.gz.dict:md5,f953d89119d6d0ae5ceab0c81aac83f6"
                    ]
                ],
                "4": [
                    "versions.yml:md5,2eaf0c4b36636d3321599dcff1b2c0ea",
                    "versions.yml:md5,3a95ae193360da51d01e9cfccdbfc74a",
                    "versions.yml:md5,69ccb03f4e3fda6002d3faca55426d2a"
                ],
                "dict": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fasta.gz.dict:md5,f953d89119d6d0ae5ceab0c81aac83f6"
                    ]
                ],
                "fai": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fasta.gz.fai:md5,9da2a56e2853dc8c0b86a9e7229c9fe5"
                    ]
                ],
                "fasta_gz": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fasta.gz:md5,6e9fe4042a72f2345f644f239272b7e6"
                    ]
                ],
                "gzi": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fasta.gz.gzi:md5,7dea362b3fac8e00956a4952a3d4f474"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2eaf0c4b36636d3321599dcff1b2c0ea",
                    "versions.yml:md5,3a95ae193360da51d01e9cfccdbfc74a",
                    "versions.yml:md5,69ccb03f4e3fda6002d3faca55426d2a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-31T09:54:00.831095632"
    }
}